src/selectors/profile.js

```diff
@@ -447,3 +451,43 @@ export const getHiddenTrackCount: Selector<HiddenTrackCount> = createSelector(
     return { hidden, total };
   }
 );
+
+export const getRelevantPagesForActiveTab: Selector<
+  Set<InnerWindowID>
+> = createSelector(
+  getPageList,
+  UrlState.getShowTabOnly,
+  (pageList, showTabOnly) => {
+    if (pageList === null || pageList.length === 0 || showTabOnly === null) {
+      // Return an empty set if we want to see everything or that data is not there.
+      return new Set();
+    }
+
+    const findRootPage = page => {
+      if (page.embedderInnerWindowID !== 0) {
+        const parent = pageList.find(
+          p => p.innerWindowID === page.embedderInnerWindowID
+        );
+        if (parent) {
+          return findRootPage(parent);
+        }
+      }
+      return page;
+    };
```

Comment:
The algorithm doesn't seem good enough to me. It's basically n^2 because for each page in pageList we look in pageList again. It's OK if we know that pageList is always small, but can we assume that?
Can we order pageList in such a way that we know that parents immediately come before or after (like we do for our other structures) so that we don't need to look at it completely? Otherwise it might be a good idea to use a Map with innerWindowID as key so that we loop the list only once, and then the lookup is constant time.
The strategy depends greatly on the size of the pageList structure obviously.

Also this selector could use some comments, explaining better the structure of the list and its content (that it's actually a tree, etc).

Reply:
@canova does this comment still need handling?